xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java

```diff
@@ -162,16 +164,18 @@ public List<DefaultNotificationFilterPreference> getPreferencesOfWiki(WikiRefere
      * @since 14.4.1
      * @since 13.10.7
      */
-    public Set<NotificationFilterPreference> getPaginatedFilterPreferences(int limit, int offset)
+    public Set<DefaultNotificationFilterPreference> getPaginatedFilterPreferences(int limit, int offset)
```

Comment:
why changing this signature?

Reply:
Because if need to have access to `getOwner` which is only available for `DefaultNotificationFilterPreference`, and not `NotificationFilterPreference`.

Reply:
Note that it's documented in the body of the method. Do you think it's interesting to add something in the javadoc too?

Reply:
No I  think it's good like that, I'm actually wondering if we shouldn't just put the owner concept in `NotificationFilterPreference` directly. AFAICS we always use a `DefaultNotificationFilterPreference` in the store API in order to set the owner, so it could almost make sense to always have it defined maybe. 
But not something to do now.